platform/o.n.bootstrap/src/org/netbeans/ProxyClassLoader.java

```diff
@@ -468,28 +468,31 @@ protected Package getPackage(String name) {
      * @return located package, or null
      */
     protected Package getPackageFast(String name, boolean recurse) {
-        synchronized (packages) {
+        if (!recurse) {
+            return null;
+        }
+        synchronized(packages) {
             Package pkg = packages.get(name);
             if (pkg != null) {
                 return pkg;
             }
-            if (!recurse) {
-                return null;
-            }
```

Comment:
this alters the behavior. The recursive check has to be after the first `get` and `return`.

I think you could do this however:

```java
        Package pkg = packages.get(name);
        if (pkg != null) {
            return pkg;
        }
        if (!recurse) {
            return null;
        }
        synchronized(packages) {
            pkg = packages.get(name);
            ...
        
```
essentially calling `get` twice, with a lock free path for the non-recursive case